packages/cdk-ecs-codedeploy-resources/lib/index.ts

```diff
@@ -0,0 +1,268 @@
+import * as cdk from "@aws-cdk/core";
+import * as customresource from "@aws-cdk/custom-resources";
+import * as iam from "@aws-cdk/aws-iam";
+import * as s3 from "@aws-cdk/aws-s3";
+import * as cloudfront from "@aws-cdk/aws-cloudfront";
+
+export interface CdkEcsCodedeployResourcesProps {
+  stage: string;
+  clusterName: string;
+  serviceName: string;
+  appName: string;
+  listenerArn: string;
+  targetGroupName: string;
+  commitHash: string;
+}
+
+export class CdkEcsCodedeployResources extends cdk.Construct {
+
+  constructor(scope: cdk.Construct, id: string, props: CdkEcsCodedeployResourcesProps) {
+    super(scope, id);
+
+    const iamRole = iam.Role.fromRoleArn(this, "EcsBlueGreenDeploymentResourcesRole", "arn:aws:iam::022393549274:role/AWSCustomResourceRoleFullAccess")
+
+    var terminationTimeout: number = props.stage.includes('prod') ? 120 : 0;
+
+    const bucket = new s3.Bucket(this, 'Bucket', {
+      bucketName: `deploy-${props.appName}.${props.stage}.ecs`,
+      publicReadAccess: true,
+      versioned: true,
+    });
+
+    const originAccessIdentity = new cloudfront.OriginAccessIdentity(this, 'OriginAcessIdentity', {
+      comment: `OriginAccessIdentity for ${bucket.bucketName}.`
+    });
+
+    const bucketPolicy = new s3.BucketPolicy(this, 'BucketPolicy', {
+      bucket: bucket
+    });
+
+    bucketPolicy.document.addStatements(
+      new iam.PolicyStatement({
+        effect: iam.Effect.ALLOW,
+        principals: [originAccessIdentity.grantPrincipal],
+        actions: ['s3:GetObject'],
+        resources: [bucket.bucketArn + "/*"],
+      })
+    );
+
+    let output = new cdk.CfnOutput(this, "AppspecBucket", {
+      value: bucket.bucketName
+    });
+
+    let output2 = new cdk.CfnOutput(this, "OriginAccessIdentity", {
+      value: originAccessIdentity.originAccessIdentityName
+    });
+
+    output.overrideLogicalId("AppspecBucket");
+    output2.overrideLogicalId("OriginAccessIdentity");
+
+    let codedeployCreateApplicationCall: customresource.AwsSdkCall = {
+      service: 'CodeDeploy',
+      action: 'createApplication',
+      parameters: {
+        applicationName: `${props.stage}-${props.appName}-CodeDeployApplication`,
```

Comment:
This `applicationName` is repeated enough that you should define it in one place and use it across all your custom resources.